### DISCUSSION TITLE
Support ESI/SSI includes

### DISCUSSION THREAD

**Discussion by timkelty** [craftcms/cloud#50] [Feature Requests & Enhancements]
Supporting Edge-side-includes would allow templates with dynamic content to still be cacheable.

**Comment by timkelty** [comment]
ESIs are now generally available for all Craft Cloud projects.
More details here: https://craftcms.com/blog/cloud-updates

**Reply by andrewfairlie** [reply]
We've got a project that could really benefit from ESI for performance. Is there any documentation for this?

**Reply by timkelty** [reply]
Nothing public yet, but plan on having something in next week or so.

FWIW, out implementation is RFC compliant with the [ESI 1.0 spec](https://www.w3.org/TR/esi-lang/), so there really isn't anything Craft or Craft Cloud specific here.

To make them easier to work with, we plan on providing a Twig tag that can fall back to regular `include` when not in a Craft Cloud environment (e.g. working locally).

I'll post some rough examples here today if you want to start playing around before we have official docs.

**Reply by timkelty** [reply]
@andrewfairlie sorry for the wait! Here's a rough example of how you can use ESI's: https://gist.github.com/timkelty/f9553c457abd37bcc3b31253e6301d60

Note: we'll soon be providing some helpers so you won't need to do the kind of include fallback I'm showing in the example to work locally.
